fix: List/price percentage ratio dynamic product groups not working (#13012)
shopware/platform: changelog/_unreleased/2025-10-14-fix-dynamic-product-group-percentage-accessor.md

---
title: Fix percentage ratio dynamic product groups
issue: 12996
---
# Core
* Changed `Shopware\Core\Content\Product\DataAbstractionLayer\CheapestPrice\CheapestPriceAccessorBuilder::buildAccessor` to ignore zero-valued entries so dynamic product group conditions based on percentage ratios evaluate correctly again.

> * Added the `symfony/polyfill-php85` dependency and moved affected internals to `array_last()` so pointer handling stays consistent across supported PHP versions.

Reviewer: ```suggestion
* Added the `symfony/polyfill-php85` dependency to make it possible to use PHP 8.5 features
```

___
# Upgrade Information

> ## array_last helper availability

Reviewer: ```suggestion
## Added PHP 8.5 polyfill
```

> The new dependency `symfony/polyfill-php85` was added, to make it possible to already use PHP 8.5 features, like `array_first` and `array_last

Reviewer: ```suggestion
The new dependency `symfony/polyfill-php85` was added, to make it possible to already use PHP 8.5 features, like `array_first` and `array_last`
```

> All installations now ship with `symfony/polyfill-php85`, which exposes the native `array_last()` helper on PHP 8.2 and 8.3. Plugin developers can drop custom implementations and rely on the global `array_last()` function for consistent pointer handling across Shopware and their extensions.

Reviewer: ```suggestion
The new dependency `symfony/polyfill-php85` was added, to make it possible to already use PHP 8.5 features, like `array_first` and `array_last`.
```

I would not necessarily mention your refactors.
